app/services/os_version_updater.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# A service class to update OS version for each policy with assigned hosts
+class OsVersionUpdater
+  class << self
+    def run!
+      Policy.all.each do |policy|
```

Comment:
find_each auto-batches by 1000 records at a time. Calling `all` tries to create an array in memory of all policies in the DB:

```suggestion
      Policy.find_each do |policy|
```